cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -40,61 +48,55 @@ newtype LocalStateQueryExpr block point query r m a = LocalStateQueryExpr
   { runLocalStateQueryExpr :: ContT (Net.Query.ClientStAcquired block point query m r) m a
   } deriving (Functor, Applicative, Monad, MonadIO)
 
--- | Use 'sendMsgQuery' in a do block to construct monadic local state queries.
-sendMsgQuery :: Monad m => query a -> LocalStateQueryExpr block point query r m a
-sendMsgQuery q = LocalStateQueryExpr . ContT $ \f -> pure $
-  Net.Query.SendMsgQuery q $
-    Net.Query.ClientStQuerying
-    { Net.Query.recvMsgResult = f
-    }
-
 -- | Execute a local state query expression.
 executeQueryLocalState
   :: LocalNodeConnectInfo mode
   -> Maybe ChainPoint
-  -> (NodeToClientVersion -> LocalStateQueryExpr (BlockInMode mode) ChainPoint (QueryInMode mode) () IO a)
-  -> IO (Either AcquireFailure a)
-executeQueryLocalState connectInfo mpoint f = do
-  resultVarQueryTipLocalState <- newEmptyTMVarIO
-  waitResult <- pure $ readTMVar resultVarQueryTipLocalState
+  -> (AcquireFailure -> e)
```

Comment:
Can't we map the error with `firstExceptT` if we wanted to?

Reply:
Yes, we can.  But in this case the function will need to return an extra layer of `Either` in order to convey both the `AcquireFailure` which this function generates, or the `e` which the monadic expression generates.

Reply:
Okay, this is sorted.